src/test/regress/expected/failure_ddl.out

```diff
@@ -0,0 +1,1228 @@
+ (localhost,57637,100800,t,3)
+ (localhost,57637,100802,t,3)
+ (localhost,57640,100801,t,3)
+ (localhost,57640,100803,t,3)
+(4 rows)
+
+-- we should be able to revocer the transaction and
+-- see that the command is rollbacked
+SELECT recover_prepared_transactions();
+ recover_prepared_transactions 
+-------------------------------
+                             1
+(1 row)
+
+SELECT run_command_on_placements('test_table', $$SELECT count(*) FROM public.table_attrs where relid = '%s'::regclass$$) ORDER BY 1;
+  run_command_on_placements   
+------------------------------
+ (localhost,57637,100800,t,3)
+ (localhost,57637,100802,t,3)
+ (localhost,57640,100801,t,3)
+ (localhost,57640,100803,t,3)
+(4 rows)
+
+-- killing on command complete of COMMIT PREPARE, we should see that the command succeeds
+-- and all the workers committed
+SELECT citus.mitmproxy('conn.onCommandComplete(command="COMMIT PREPARED").kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+ALTER TABLE test_table DROP COLUMN new_column;
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT count(*) FROM public.table_attrs where relid = 'test_table'::regclass;
+ count 
+-------
+     2
+(1 row)
+
+SELECT run_command_on_placements('test_table', $$SELECT count(*) FROM public.table_attrs where relid = '%s'::regclass$$) ORDER BY 1;
+  run_command_on_placements   
+------------------------------
+ (localhost,57637,100800,t,2)
+ (localhost,57637,100802,t,2)
+ (localhost,57640,100801,t,2)
+ (localhost,57640,100803,t,2)
+(4 rows)
+
+-- we shouldn't have any prepared transactions in the workers
+SELECT recover_prepared_transactions();
+ recover_prepared_transactions 
+-------------------------------
+                             0
+(1 row)
+
+SELECT run_command_on_placements('test_table', $$SELECT count(*) FROM public.table_attrs where relid = '%s'::regclass$$) ORDER BY 1;
+  run_command_on_placements   
+------------------------------
+ (localhost,57637,100800,t,2)
+ (localhost,57637,100802,t,2)
+ (localhost,57640,100801,t,2)
+ (localhost,57640,100803,t,2)
+(4 rows)
+
+-- kill as soon as the coordinator sends COMMIT
+SELECT citus.mitmproxy('conn.onQuery(query="^COMMIT PREPARED").kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+ALTER TABLE test_table ADD COLUMN new_column INT;
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+-- some of the placements would be missing the new column
+-- since we've not commited the prepared transactions
+SELECT count(*) FROM public.table_attrs where relid = 'test_table'::regclass;
+ count 
+-------
+     3
+(1 row)
+
+SELECT run_command_on_placements('test_table', $$SELECT count(*) FROM public.table_attrs where relid = '%s'::regclass$$) ORDER BY 1;
+  run_command_on_placements   
+------------------------------
+ (localhost,57637,100800,t,3)
+ (localhost,57637,100802,t,3)
+ (localhost,57640,100801,t,2)
+ (localhost,57640,100803,t,2)
+(4 rows)
+
+-- we should be able to revocer the transaction and
```

Comment:
"revocer" -> "recover"

Reply:
While following along by doing it by hand I accidentally noticed that recover_prepared_transactions() probably has a bug. If I leave in the part which kills connections on `^COMMIT`:

```
brian=# SELECT recover_prepared_transactions();
WARNING:  server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
WARNING:  connection not open
 recover_prepared_transactions 
-------------------------------
                             0
(1 row)
```

It returns the number of transactions which it was able to recover, but doesn't give any indication that there are still transactions left to recover!

Reply:
Yeah, I remember this. But, aren't the `WARNING`s  important indication for the necessity to re-run the recovery?

I remember during the PR for `recover_prepared_transactions ()` we decided to warn the user about the connection errors, but, do not fail the recovery at all since we could potentially recover prepared transactions from the other workers.

Do you see any other approaches for that?

Reply:
I understand the idea but I don't agree that this implementation matches it. Why not error out here? Recovery itself has errored! Warnings are something you can ignore, here it feels like we're suppressing an important failure. Until `recover_prepared_transactions()` runs all your queries on this table are going to block and you won't know why.

This isn't part of this PR though, I guess I should open a ticket.

Reply:
>  If we want to continue and try to recover prepared transactions from the other workers, we can still try to do that before we raise the ERROR.

Yes, that would be a nice improvement.

Reply:
Just note that we've got the exact same approach with deadlock detection where we warn the user if we cannot connect one of the nodes. We should probably improve both!